File: backend/app/models/lead.py
# Modelo de tabla `leads`: un negocio pequeño de Texas con su calificación
# de presencia digital. SQLModel = SQLAlchemy + Pydantic en una sola clase.
from datetime import datetime, timezone

from sqlmodel import Field, SQLModel


def _utc_now() -> datetime:
    # Timestamps siempre en UTC (evita líos de zona horaria)
    return datetime.now(timezone.utc)


class Lead(SQLModel, table=True):
    """Small Texas business lead with online-presence qualification."""

    __tablename__ = "leads"

    id: int | None = Field(default=None, primary_key=True)      # autoincremental
    external_id: str = Field(index=True, unique=True)           # taxpayer number (clave estable entre backends)
    name: str = Field(index=True)                               # razón social
    city: str = Field(index=True)                               # ciudad
    county: str | None = None                                   # condado
    state: str = Field(default="TX", index=True)                # estado (siempre TX)
    zip_code: str | None = None                                 # código postal
    latitude: float | None = None                               # coordenadas para radio
    longitude: float | None = None
    phone: str | None = None                                    # contacto
    email: str | None = None
    industry: str | None = Field(default=None, index=True)      # giro del negocio
    employee_count: int | None = None                           # tamaño (si se conoce)
    website_url: str | None = None                              # sitio web guardado
    has_website: bool = False                                   # señales digitales ↓
    website_reachable: bool = False                             # ¿el sitio responde?
    has_modern_website: bool = False                            # ¿parece moderno?
    website_tech_stack: str | None = None                       # stack detectado
    website_antiquity_years: int | None = None                  # antigüedad estimada
    website_analysis_notes: str | None = None                   # resumen del análisis
    facebook_url: str | None = None                             # redes sociales
    instagram_url: str | None = None
    has_active_facebook: bool = False
    has_active_instagram: bool = False
    is_small_business: bool = True                              # clasificación pequeño negocio
    is_qualified: bool = Field(default=False, index=True)       # ¿lead calificado?
    qualification_score: float = Field(default=0.0)             # puntaje 0-100
    qualification_notes: str | None = None                      # razones del puntaje
    sells_alcohol: bool = Field(default=False, index=True)      # cruce TABC ↓
    alcohol_segment: str | None = None                          # bar / restaurant
    liquor_receipts_total: float | None = None                  # totales de recibos
    wine_receipts_total: float | None = None
    beer_receipts_total: float | None = None
    total_receipts_total: float | None = None
    source: str = "texas_public_stub"                           # origen del dato
    created_at: datetime = Field(default_factory=_utc_now)      # timestamps UTC
    updated_at: datetime = Field(default_factory=_utc_now)

File: frontend-v2/app/types/lead.ts
/** Un negocio (lead) tal como lo devuelve la API FastAPI (/api/leads). */
export interface Lead {
  id: number                              // id interno de la base
  external_id: string                     // taxpayer number del Comptroller de Texas
  name: string                            // razón social / nombre del negocio
  city: string                            // ciudad
  county: string | null                   // condado
  state: string                           // estado (siempre TX)
  zip_code: string | null                 // código postal
  latitude: number | null                 // coordenadas (si se geocodificó)
  longitude: number | null
  distance_miles: number | null           // distancia al centro de búsqueda por radio
  phone: string | null                    // teléfono de contacto
  email: string | null                    // email de contacto
  industry: string | null                 // industria/giro del negocio
  employee_count: number | null           // nº de empleados (si se conoce)
  website_url: string | null              // URL del sitio web detectado/guardado
  has_website: boolean                    // ¿tiene sitio web?
  website_reachable: boolean              // ¿el sitio responde?
  has_modern_website: boolean             // ¿el sitio parece moderno? (análisis)
  website_tech_stack: string | null       // tecnologías detectadas (WordPress…)
  website_antiquity_years: number | null  // antigüedad estimada del sitio (años)
  website_analysis_notes: string | null   // notas del análisis (incluye marca TABC)
  facebook_url: string | null             // redes sociales detectadas
  instagram_url: string | null
  has_active_facebook: boolean            // ¿la página de FB parece activa?
  has_active_instagram: boolean
  is_small_business: boolean              // ¿clasificado como negocio pequeño?
  is_qualified: boolean                   // ¿lead calificado? (score sobre umbral)
  qualification_score: number             // puntaje de calificación (0-100)
  qualification_notes: string | null      // por qué calificó / no calificó
  sells_alcohol?: boolean                 // cruce TABC mixed beverage
  source: string                          // origen del dato (franchise, TABC…)
  created_at: string                      // timestamps ISO
  updated_at: string
}
